schema.md

```diff
@@ -0,0 +1,136 @@
+# Sparkle Motion schema (WIP)
+
+All record types have `createdAt` (and maybe `modifiedAt`?).
+
+### User
+
+```js
+{
+  "id": ___,
+  "username": "xian",
+  "password": "...",      // salted hash of pw
+  "name": "Full Name",
+  "priority": 123.4,      // e.g. for locking out other users during a performance
+}
+```
+
+### Gadget Presets
+
+Gadgets offer the ability to save their current values as a user preset, or to load presets.
+
+```js
+{
+  "id": ___,
+  "userId": ___,
+  "name": "Pride Flag",
+  "type": "Palette",      // the type of the gadget
+  "data": { ... },        // type-dependent
+}
+```
+
+#### Color Gadget Data
+```js
+{
+  "color": 0xfedbca,
+}
+```
+
+#### Palette Gadget Data
+```js
+{
+  "type": "analogous" | "monochromatic" | "triad" | "complementary" | "compound" | "shades" | "custom",
+
+  // e.g. for type "custom":
+  "values": [0xE70000, 0xFF8C00, 0xFFEF00, 0x00811F, 0x0044FF, 0x760089],
+}
+
+```
+
+#### Slider Gadget Data
+```js
+{
+  "value": 0.7325,
```

Comment:
I think we should also include a range and step for this?

```js
{
  "value": 0.1337,
  "range": [0, 1],
  "step": 0.0001
}
```
```

Reply:
Agreed, adding.